pkg/controller/lifecycle/actuator.go

```diff
@@ -409,12 +408,10 @@ func getSeedResources(oidcReplicas *int32, hibernated bool, namespace, genericKu
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
 					Labels: utils.MergeStringMaps(getLabels(), map[string]string{
-						v1beta1constants.LabelNetworkPolicyToDNS:                v1beta1constants.LabelNetworkPolicyAllowed,
-						v1beta1constants.LabelNetworkPolicyFromShootAPIServer:   v1beta1constants.LabelNetworkPolicyAllowed,
-						v1beta1constants.LabelNetworkPolicyToShootAPIServer:     v1beta1constants.LabelNetworkPolicyAllowed,
-						v1beta1constants.LabelNetworkPolicyToPublicNetworks:     v1beta1constants.LabelNetworkPolicyAllowed,
-						v1beta1constants.LabelNetworkPolicyToPrivateNetworks:    v1beta1constants.LabelNetworkPolicyAllowed,
-						v1beta1constants.LabelNetworkPolicyToAllShootAPIServers: v1beta1constants.LabelNetworkPolicyAllowed,
```

Comment:
I am not sure that the `allow-to-all-shoot-apiservers` label is unneeded. In the past we had issues which occurred when user was trying to establish trust to an apiserver located on the same seed as its own apiserver. This is why this label was added with [this](https://github.com/gardener/gardener-extension-shoot-oidc-service/pull/23) PR. Let's not remove it for now. I can check during validation of this extension if this is still needed or not.

Reply:
After https://github.com/gardener/gardener/pull/7408/commits/06b8c434a791d3819bbe334cba2202b25736c3c3, (released with `gardener/gardener@v1.64`), this `NetworkPolicy` is no longer created in the shoot namespaces. Hence, the label is useless for now.
I am also wondering why this was needed in the first place. For the scenario you described, the network path should always be

- `kube-apiserver-of-shoot-A` -> `internet` -> `istio-ingressgateway` -> `kube-apiserver-of-shoot-B`

instead of a direct connection from `kube-apiserver-of-shoot-A` -> `kube-apiserver-of-shoot-B`.

Reply:
I remember that we investigated this issue with @ScheererJ and the behavior was indeed not expected.

> For the scenario you described, the network path should always be
`OWA-shoot-A -> internet -> istio-ingressgateway -> kube-apiserver-of-shoot-B`

Yes, this was also my assumption back then. Instead it was rather `OWA-shoot-A -> kube-apiserver-of-shoot-B` when using the internal domain of shoot-B hence I added the network policy label. Anyway, this is probably not issue regarding this PR since it was already removed and I should check this behavior separately.

I cannot find a release note regarding [this](https://github.com/gardener/gardener/commit/06b8c434a791d3819bbe334cba2202b25736c3c3) change in the 1.64 release notes. What will be the correct path forward in case the said network policy is needed? Should the extension deploy it on its own and handle all labels/selectors?

Reply:
Well, https://sap-ti.slack.com/archives/G0177CM9ZAP/p1678183320567479?thread_ts=1678175235.133789&cid=G0177CM9ZAP seems to explain this networking behaviour.

I would consider it a bug that the `to-all-shoot-apiservers` ever existed in shoot namespaces themselves. This happened unintentionally and was never supposed to be an "API"/contract extensions can rely on - hence, no release note for the removal. 

> What will be the correct path forward in case the said network policy is needed? Should the extension deploy it on its own and handle all labels/selectors?

- One approach would be extending the namespace selectors in the `networking.resources.gardener.cloud/namespace-selectors` annotation of `kube-apiserver` services with `{"matchLabels":{"gardener.cloud/role":"shoot"}}`. This would allow to use `networking.resources.gardener.cloud/to-all-shoots-kube-apiserver-tcp-443=allowed` label on pods in the shoot namespaces.
- Another approach (maybe easier) is to deploy a `NetworkPolicy` to shoot namespaces which allows egress to all API servers in all shoot namespaces for the OIDC pods. I guess this is easier/more sustainable for this extension. Do you want me to do it as part of this PR?

Reply:
Thanks for the explanation. I prefer that I open a separate PR for this change and I will invite you for a review. Please adapt the [compatibility matrix](https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/master/README.md#compatibility) to reflect which version of Gardener is needed before upgrading to this version of the extension and I guess we are good to go with this PR.

Reply:
Done, PTAL. The assumption is that you'll fix above scenario before releasing `v0.16`.